packages/core/lib/messages/baseMessageSchemas.ts

```diff
@@ -0,0 +1,25 @@
+import z from 'zod'
+
+export const BASE_MESSAGE_SCHEMA = z.object({
+  id: z.string().describe('event unique identifier'),
+  type: z.literal<string>('<replace.me>').describe('event type name'),
+  payload: z.optional(z.object({})).describe('event payload based on type'),
+})
+
+export const EXTENDED_MESSAGE_SCHEMA = BASE_MESSAGE_SCHEMA.extend({
+  metadata: z
+    .object({
+      schemaVersion: z.string().min(1).describe('base event schema version'),
+      timestamp: z.string().datetime().describe('iso 8601 datetime'),
+      producerApp: z.string().min(1).describe('app/service that produced the event'),
```

Comment:
does this field raise the visibility somehow? we will see this information only in the logs potentially

Reply:
addressed